lib/FlyBy.pm

```diff
@@ -255,7 +255,11 @@ sub all_keys {
 
 sub values_for_key {
     my ($self, $key) = @_;
-    return (sort { $a cmp $b } keys %{$self->index_sets->{$key}});
+
+    my $set = $self->index_sets->{$key};
+    my @values = grep { my $v = $set->{$_}; $v && %{$v->{elements}} } keys %$set;
```

Comment:
That might work but it involves undocumented internals of the [Set::Scalar](https://metacpan.org/pod/Set::Scalar) module. I don't like it.